[Tickets] Profile Links & Pending Count

DonovanDMC opened the pull request:
This pr:
* Adds a ticket count to user's profile, visible to them and moderator+ (along with pending & accused links)
* Displays the "Accused" user within tickets, along with a count of the "active" tickets against them (pending + partial)
* Replaces the timestamps with the standard `time_ago_in_words_tagged`

Also due to editor magic ✨, I've fixed all of the `Style/HashSyntax` issues in the user presenter. (..it's almost too tempting to go around fixing every tiny error)
The important lines are 137-139.

DonovanDMC committed:
[Tickets] Profile Links & Pending Count

DonovanDMC committed:
stuff a sock in it, rubocop

Earlopain commented on app/models/user.rb:
```diff
@@ -762,7 +766,8 @@ def refresh_counts!
           post_update_count: PostVersion.for_user(id).count,
-          note_count: NoteVersion.where(updater_id: id).count
+          note_count: NoteVersion.for_user(id).count,
+          ticket_count: Ticket.for_user(id).count,
```
I see not reason to fix something that isn't broken in the first place. I have not found evidence that the status counter gets out of sync

DonovanDMC replied:
With something that will be counted so infrequently, you are probably correct
I had assumed that *everything* would be included in a count refresh, regardless of how infrequent or unlikely they are to be inaccurate

Earlopain commented on app/views/tickets/index.html.erb:
```diff
@@ -50,8 +50,8 @@
             <% end %>
 
             <td class="<%= ticket.status %>-ticket"><%= pretty_ticket_status(ticket) %></td>
-            <td style="cursor:help;" title="<%= ticket.updated_at.strftime("%b %d, %Y %I:%M %p") %>"><%= time_ago_in_words(ticket.updated_at) %> ago</td>
```
Just use the way every other place display time, there's a helper method for that. That would be the proper way

DonovanDMC replied:
> I would have used `time_ago_in_words_tagged` for [this], but that would slightly change the way it works currently. Right now, the title is on the entire column.

If we don't care about that difference, that's all good and well, I just didn't want to change something that might have been a specific way just to have to undo it later

Earlopain replied:
I can't imagine any reason why it should be the way it is currently specifically. Kira was probably just in a rush to get it all working. Sets/Takedowns don't adhere to the general style of the codebase as well. That results in these slight discrepancies with the time formatting you noticed.

DonovanDMC replied:
Standardization gets no complaints from me

DonovanDMC committed:
Remove counts fixer for ticketss

Earlopain commented on app/views/tickets/show.html.erb:
```diff
@@ -71,10 +80,10 @@
         </tr>
 
-        <% if(!@ticket.response.blank?) %>
+        <% unless @ticket.response.blank? %>
```
I'm not really a fan of `unless` if it can be avoided.

```suggestion
        if @ticket.response.present?
```

Earlopain commented on app/views/tickets/show.html.erb:
```diff
@@ -71,10 +80,10 @@
           <tr>
             <td><span class='title'>Handled by</span></td>
-            <% if (!@ticket.handler.nil?)%>
+            <% if !@ticket.handler.nil? %>
```
```suggestion
            <% if @ticket.handler.present? %>
```

DonovanDMC committed:
Update app/views/tickets/show.html.erb

Co-authored-by: Earlopain <earlopain+github@protonmail.com>

DonovanDMC committed:
Update app/views/tickets/show.html.erb

Co-authored-by: Earlopain <earlopain+github@protonmail.com>

DonovanDMC committed:
use standardized timestamps

Earlopain commented on app/views/tickets/show.html.erb:
```diff
@@ -83,7 +83,7 @@
           <tr>
             <td><span class='title'>Handled by</span></td>
-            <% if !@ticket.handler.nil? %>
+            <% if @ticket.handler.present? %>
```
Quick note, you were previously saying you wanted to do a nil check to prevent a database query. I checked and rails is clever enough to not do a database query in this case. It doesn't matter if you do `handler_id.nil?` or go directly for `handler.present?`

DonovanDMC replied:
The only time I can remember saving a db call is checking the qtype in https://github.com/e621ng/e621ng/pull/514#discussion_r1189490075

Still, nice to know

Earlopain replied:
Yeah, that's what I was talking about. Same situation here

DonovanDMC committed:
this got yeeted at some point

Earlopain committed:
Fix search links, only show pending count when >= 2

I'm imagining some confusion if it shows one pending when it is just
the one you're looking at right now

DonovanDMC committed:
Fix fixer? `update_column` throws a readonly error

DonovanDMC committed:
`for_user` is only used in fixer script

Earlopain committed:
Fix fixer script, for real
